cql3/expr/prepare_expr.cc

```diff
@@ -1000,9 +1000,39 @@ try_prepare_expression(const expression& expr, data_dictionary::database db, con
                 if (!prepared_arg_opt) {
                     return std::nullopt;
                 }
+
                 prepared_token_args.emplace_back(std::move(*prepared_arg_opt));
             }
 
+            // Validate that the token() function was passed all partition key columns as the arguments.
```

Comment:
Again, I think we shouldn't expect the partition key columns themselves, but just the right types.

Reply:
By the way, this is exactly why I keep preaching about cql-pytest being a better test platform than C++ tests: Here (unless I'm mistaken and you are right! please correct me if I'm wrong!), you misunderstood the requirement (thought that you must list the columns, not just the right types), and wrote the wrong code code plus the wrong test to match it. cql-pytest would have allowed you to run the same test against Cassandra, and realize that you set the wrong goal for the code and the test.

Reply:
By the way, if I made it sound like the only value of cql-pytest is the ability to compare to **Cassandra** I misspoke. Not only does it let you compare to Cassandra, it lets you compare to older versions of Scylla - something which is very important in the case of refactoring like you're doing now (by the way, Kostja's textual "cql" suite also has this feature). It is possible that in earlier Scylla the CQL expression "token(1, 2)" worked fine when the partition key is two integer columns and that your new code will break it - but we'll not know this based on the C++ test because the C++ test can only check the new version (with expressions), not the old version.

All that being said, there's nothing wrong with having *more* tests - like both C++ and cql-pytest tests. Just in this case I believe the C++ tests were simply wrong (and you would have discovered it when token() started to use the new code and failed the cql-pytest tests - and at that point you would have fixed both the code and the C++ tests).